### DISCUSSION TITLE
Ayuda! Configuracion oyente HTTP, Token, conversion de JSON a HL7

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#5375] [Q&A]
Buenas, les comento que estoy iniciando en el uso de Mirth, he revisado material y videos. Sin embargo no me esclarece mucho la informacion.
Necesito un crear un canal de oyente HTTP, configurar token, y generar conversion del archivo que llegaria en formato JSON, debo alojarlo en carpeta local con formato HL7.

Agradeceria una guia clara, o informacion precisa.
Se que es mucho, pero tengo muchas ganas de aprender y lo estoy haciendo de forma totalmente autodidacta.

**Comment by jonbartels** [comment]
You have to break the problem down into smaller steps.

1. Set up a test in Postman or `curl` - Make it POST the JSON you want to an HTTP endpoint
2. Write a channel that is an HTTP listener that accepts JSON and has a Channel Writer in the RAW format - run your test script, do you get JSON?
3. Change your channel writer to be the HL7 data type. Get it so that when you receive JSON it writes an empty HL7 template (hint - read about "outbound message template") - run your test script, do you get an empty HL7 message in the channel writer?
4. Change your Channel Writer to a file writer, test it again, do you get a file written out?
5. Read some fields from your JSON message and insert them into the HL7 template. Test it, you should see that data getting populated
6. Repeat step 5 and keep adding a few mappings at a time.
7. Now worry about authentication. What kind of authn do you want to use?

**Reply by 5011-5011** [reply]
Muchas gracias por la orientacion de iniciar el canal y sus pruebas. Hare hasta el punto 6, una vez logrado eso le comento para lo del token.
Gracias nuevamente por su tiempo y disposicion.
